src/torrent.cpp

```diff
@@ -7543,22 +7542,38 @@ namespace {
 			return false;
 		}
 
-		// now, we might already have this torrent in the session.
-		m_torrent_file->info_hashes().for_each([&](sha1_hash const& ih, protocol_version)
+		// we might already have this torrent in the session.
+		bool failed = false;
+		info->info_hashes().for_each([&](sha1_hash const& ih, protocol_version)
 		{
+			if (failed) return;
+
 			auto t = m_ses.find_torrent(info_hash_t(ih)).lock();
 			if (t && t != shared_from_this())
 			{
-				// TODO: if the existing torrent doesn't have metadata, insert
-				// the metadata we just downloaded into it.
+				TORRENT_ASSERT(!t->valid_metadata());
+
+				// if we get a collision, both torrents fail and have to be
+				// removed.
```

Comment:
It may be helpful for future debuggers to link to the github issue to figure out the author's thought process